packages/js-client/src/types.ts

```diff
@@ -59,11 +90,22 @@ type ExtractPathAndQueryParameters<K extends keyof operations> = 'parameters' ex
       : undefined
   : undefined
 
-type OperationParams<K extends keyof operations> = 'parameters' extends keyof operations[K]
-  ? IsPathAndQueryOptional<K> extends true
-    ? ExtractPathAndQueryParameters<K> | void
+type CombinedParamsAndRequestBody<K extends keyof operations> =
+  HasRequestBody<K> extends true
+    ? ExtractPathAndQueryParameters<K> & {
+        body: RequestBody<K>
+      }
     : ExtractPathAndQueryParameters<K>
-  : void
+
+type OperationParams<K extends keyof operations> = 'parameters' extends keyof operations[K]
+  ? IsAnythingRequired<K> extends false
+    ? CombinedParamsAndRequestBody<K> | void
+    : CombinedParamsAndRequestBody<K>
+  : HasRequestBody<K> extends true
+    ? IsRequestBodyOptional<K> extends true
+      ? { body: RequestBody<K> } | void
+      : { body: RequestBody<K> }
+    : void
```

Comment:
todo: probably move this `IsRequestBodyOptional` check up into the `IsAnythingRequired` type.